exp/views/study.py

```diff
@@ -913,11 +918,26 @@ def can_preview(self):
 
     test_func = can_preview
 
+    def get(self, request, *args, **kwargs):
+        # Need to check for AWS variables here instead of get_context_data, so that we can redirect if there's an error.
+        aws_vars = get_jspsych_aws_values()
+        if aws_vars is None:
+            messages.error(
```

Comment:
You can just save the `aws_vars` to `self.aws_vars`.  It is probably not great to allocate memory for this variable in two spaces.

Reply:
Doh! Yep. Thanks for that - fixed!